### PR TITLE
Basic dropdown

### PR DESCRIPTION
**Pull request by freeman14**

### CODE REVIEW

File: src/directives/dropdown/dropdown.directive.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+          // Handle click only on trigger button
+          element.bind('click', (event) => {
+            if (event.target === element[0]) {
+              element.toggleClass('dd__dropdown--open');
+            }
+          });
+
+          // Handle body click
+          angular.element(document.body).bind('click', (event) => {
```
Comment (kurpav) [inline src/directives/dropdown/dropdown.directive.ts]: add `scope.$on('$destroy')` and release all resources

File: src/styles/blocks/dropdown.less [hunk 2]
```diff
@@ -154,3 +154,29 @@
...
+}
+
+.dd__dropdown--hover:hover .dd__dropdown--body{
+  display: block;
+}
+
+.dd__dropdown--body {
+  display: none;
+  background-color: @white;
+	box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
```
Comment (kurpav) [inline src/styles/blocks/dropdown.less:176]: As far as I remember there is no autoprefixer here, so please use mixins

**Review by kurpav — CHANGES_REQUESTED** [review]

File: src/styles/blocks/dropdown.less (hunk 2 above)
Comment (freeman14) [reply src/styles/blocks/dropdown.less:176]: https://css-tricks.com/do-we-need-box-shadow-prefixes/

File: src/directives/dropdown/dropdown.directive.ts [hunk 3]
```diff
@@ -37,7 +29,20 @@ export function DropdownElement(): any {
 
             }
 
+          };
+
+          // Handle click only on trigger button
+          element.bind('click', (event) => {
```
Comment (kurpav) [inline src/directives/dropdown/dropdown.directive.ts]: I guess we should remove this listener as well.

Comment (freeman14) [inline src/directives/dropdown/dropdown.directive.ts]: ?

**Review by kurpav — APPROVED** [review]

File: src/directives/dropdown/dropdown.directive.ts (hunk 3 above)
Comment (freeman14) [reply src/directives/dropdown/dropdown.directive.ts]: fixed

**Pull request merged**